File: statuserver/statuserver/statuserver/server/importData.ts
import type { IStorage } from "./storage";
import type { InsertService } from "@shared/schema";

interface ImportedService {
  Name: string;
  Type: string;
  Environment: string;
  Address: string;
  Port: number;
}

interface ImportData {
  [environment: string]: {
    [category: string]: ImportedService[];
  };
}

export async function importServicesFromData(storage: IStorage, data: ImportData) {
  const services: InsertService[] = [];
  
  for (const environment of Object.keys(data)) {
    for (const category of Object.keys(data[environment])) {
      const items = data[environment][category];
      
      for (const item of items) {
        const service: InsertService = {
          name: item.Name,
          description: `${item.Type} service`,
          category: category,
          region: environment,
          status: "operational",
          type: item.Type,
          icon: getIconForType(item.Type),
          address: item.Address,
          port: item.Port,
        };
        
        services.push(service);
      }
    }
  }
  
  for (const service of services) {
    await storage.createService(service);
  }
  
  return services.length;
}

function getIconForType(type: string): string {
  const typeMap: Record<string, string> = {
    'DevTools': 'wrench',
    'Backend': 'server',
    'Frontend': 'globe',
    'BPMN': 'workflow',
    'PSQL': 'database',
    'Database': 'database',
    'Keycloak': 'shield',
    'Grafana': 'chart',
    'Kafka': 'message-square',
    'Minio': 'database',
    'Redis': 'database',
    'RabbitMQ': 'message-square',
  };
  
  return typeMap[type] || 'server';
}

File: statuserver/server_py/import_data.py
from typing import Dict, List, Any
from models import InsertService

def get_icon_for_type(service_type: str) -> str:
    type_map = {
        'DevTools': 'wrench',
        'Backend': 'server',
        'Frontend': 'globe',
        'BPMN': 'workflow',
        'PSQL': 'database',
        'Database': 'database',
        'Keycloak': 'shield',
        'Grafana': 'chart',
        'Kafka': 'message-square',
        'Minio': 'database',
        'Redis': 'database',
        'RabbitMQ': 'message-square',
    }
    return type_map.get(service_type, 'server')

async def import_services_from_data(storage, data: Dict[str, Dict[str, List[Dict[str, Any]]]]) -> int:
    services: List[InsertService] = []
    
    for environment in data.keys():
        for category in data[environment].keys():
            items = data[environment][category]
            
            for item in items:
                service = InsertService(
                    name=item.get('Name', ''),
                    description=f"{item.get('Type', '')} service",
                    category=category,
                    region=environment,
                    status="operational",
                    type=item.get('Type'),
                    icon=get_icon_for_type(item.get('Type', '')),
                    address=item.get('Address'),
                    port=item.get('Port')
                )
                services.append(service)
    
    for service in services:
        await storage.create_service(service)
    
    return len(services)
